src/main/java/frc/robot/commands/AutoShoot.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.Constants;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.subsystems.GoalCamera;
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.Shooter;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+
+public class AutoShoot extends CommandBase {
+
+  private Drivetrain drivetrain;
+  private GoalCamera goalCamera;
+  private double robotVelocity;
+  private Command runIntake;
+  private Command runFlywheel;
+  private Trajectory finalTrajectory;
+
+  public AutoShoot(Drivetrain drivetrain, Intake intake, Shooter shooter, GoalCamera goalcamera, double robotVelocity, FollowTrajectory followTrajectory) {
+    //defining classes
+    this.drivetrain = drivetrain;
+    this.goalCamera = goalcamera;
+    this.robotVelocity = robotVelocity;
+    this.runFlywheel = new RunFlywheel(shooter, Constants.ksVolts);
+    this.runIntake = new RunIntake(intake, Constants.defaultIntakeSpeed);
+
+  }
+
+
+  // Called just before this Command runs the first time
+  @Override
+  public void initialize() {
+    if (finalTrajectory != null){
+      this.schedule();
+    }
+
+  
+    
+  }
```

Comment:
The initialize method is the first thing that will run when this command is scheduled. Here is a basic order of events and what that should look like.
- Store trajectory from goal camera in object variable (drivetrain velocity can be found in drivetrain subsystem)
- If that trajectory is null (meaning no target detected) don't proceed with the following steps.
- Using the trajectory create a FollowTrajectory Command.
- Store a command group that executes the FollowTrajectory Command and then executes the shoot sequence while locking down the drivetrain.
-Schedule that command group.

Reply:
Ok thank you for the feedback. The issue makes a lot more sense now that I realize that everything in the command is supposed to happen in the initialize method. I will work on streamlining the command so that everything occurs in the initialize method and return the entire conditional wherever there is an if statement. I should be able to commit these changes by the end of the day tomorrow.